packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -473,16 +496,23 @@ class _DropdownMenuState extends State<DropdownMenu> {
                   controller: _textEditingController,
                   onEditingComplete: () {
                     if (currentHighlight != null) {
-                      _textEditingController.text = filteredEntries[currentHighlight!].label;
+                      final DropdownMenuEntry<T> entry = filteredEntries[currentHighlight!];
+                      _textEditingController.text = entry.label;
                       _textEditingController.selection =
-                          TextSelection.collapsed(offset: _textEditingController.text.length);
+                        TextSelection.collapsed(offset: _textEditingController.text.length);
+                      widget.onSelected?.call(entry.value);
```

Comment:
Would it make sense to call `onSelected` with null here, when there is no selection?

Reply:
For here, because the `currentHighlight` is not null, so it means the user has highlighted one item, and pressed 'enter' because it is in the `onEditingComplete`. So I just assume that this part can be considered as a selection.

Reply:
I get that, and that makes sense. However if they have hit enter on text that doesn't match an entry (i.e. `currentHighlight` is null), wouldn't we want to call the `onSelected` with null as there isn't a valid selection but they are done editing?